### PR TITLE
PRSD-453: One Login header

### PR DESCRIPTION
**Pull request by isobel-softwire**
### Features
- Creates `authenticatedHeader` fragment based on [One Login Header](https://github.com/govuk-one-login/service-header)
- Creates `header` fragment that shows the `authenticatedHeader` if the user if the user is authenticated and the `anonymousHeader` (the original header) if the user isn't
- Adds a placeholder `/signout` endpoint for 'Sign Out' to point to
- Adds example navigation links to `/registration`

### Screenshot
image

### CODE REVIEW

**Comment by isobel-softwire** [comment]
There are line terminator symbol changes in the layout and anonymousHeader fragments so it looks like there are more changes than there actually are.

File: rollup.config.mjs [hunk 1]
```diff
@@ -35,6 +35,14 @@ export default {
                         src: 'node_modules/govuk-frontend/dist/govuk/assets/manifest.json',
                         dest: 'dist',
                     },
+                    {
+                        src: 'node_modules/govuk-one-login-service-header/dist/scripts/service-header.js',
```
Comment (Travis-Softwire) [inline rollup.config.mjs:39]: I haven't looked at the library, but have they not managed to set this up as a proper node module that we can import in index.js?

File: src/main/resources/templates/fragments/header/header.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<html xmlns:sec="http://www.thymeleaf.org/extras/spring-security">
+<div th:fragment="header(serviceName, navLinks)">
```
Comment (Travis-Softwire) [inline src/main/resources/templates/fragments/header/header.html:2]: Does this mean we always need to provide a navLinks array even for pages where the user isn't logged in?

Comment (isobel-softwire) [reply src/main/resources/templates/fragments/header/header.html:2]: No, null is being passed when the `header` fragment is called (from `layout`) if the model doesn't provide an array.

**Review by Travis-Softwire — APPROVED** [review]
Left a Q about how we import the module from the 1L header library but that aside LGTM :)

File: rollup.config.mjs (hunk 1 above)
Comment (isobel-softwire) [reply rollup.config.mjs:39]: There doesn't seem to be

**Pull request merged**